third_party/prost/private/protoc_wrapper.rs

```diff
@@ -0,0 +1,404 @@
+        let package = PathBuf::from(file_path.trim_matches('.'));
+
+        let package_path_str = package
+            .to_str()
+            .context("Failed to convert package path to string")?
+            .to_lowercase()
+            .replace('.', "/")
+            .to_owned();
+
+        if package_path_str.is_empty() {
+            continue;
+        }
+
+        let mut package_path = PathBuf::from(&package_path_str);
+        if module_info.contains_key(&package_path_str) {
+            return Err(anyhow!(
+                "Multiple files generated for package {package_path_str}",
+            ));
+        }
+
+        module_info.insert(
+            package_path_str.to_string(),
+            fs::read_to_string(path).context("Failed to read file")?,
+        );
+
+        while let Some(parent) = package_path.parent() {
+            let parent_str = parent
+                .to_str()
+                .context("Failed to convert parent to string")?;
+            if parent_str.is_empty() {
+                break;
+            }
+            if let Some(existing) = module_info.insert(parent_str.to_string(), "".to_string()) {
+                // Restore any existing content
+                if !existing.is_empty() {
+                    module_info.insert(parent_str.to_string(), existing + "\n");
+                }
+            }
+            package_path = parent.to_path_buf();
+        }
+    }
+
+    let mut content = "// @generated\n\n".to_string();
+
+    let mut closing = String::new();
+    for (path, rs_content) in module_info {
+        let module = path
+            .rsplit_once('/')
+            .map(|tup| tup.1)
+            .unwrap_or(&path)
+            .to_lowercase();
+
+        content += &format!("pub mod {} {{\n", module);
+        content += &rs_content;
+
+        closing += "}\n";
+        if !rs_content.is_empty() {
+            content += &closing;
+            closing = "".to_owned();
+        }
+    }
+
+    Ok(content + "\n")
+}
+
+/// Compute the `--extern_path` flags for the given prost outputs.
+///
+/// This is expected to convert a file `.example.prost.helloworld.rs` to
+/// `.example.prost.helloworld=crate_name::example::prost::helloworld`
+fn compute_proto_package_info(prost_output: &Path, crate_name: &str) -> Result<String> {
+    let mut package = prost_output
+        .file_stem()
+        .context("Failed to get prost output file stem")?
+        .to_str()
+        .context("Failed to convert prost output file stem to string")?
+        .to_string();
+
+    if cfg!(feature = "tonic") {
+        package = package
+            .strip_suffix(".tonic")
+            .context("Failed to strip tonic suffix from package name")?
+            .to_string();
+    }
+
+    let module = format!(
+        "{}::{}",
+        crate_name,
+        package.to_lowercase().replace('.', "::")
+    );
+
+    Ok(format!(".{}={}", package, module))
+}
+
+#[derive(Debug, Parser)]
```

Comment:
I actually think this should not use `clap` so it's a lighter weight tool for consumers of the rule.